src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(filterMen);
+  const result = men.reduce(averageAge, 0);
+
+  function filterMen(person) {
+    return century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm';
+  }
+
+  function averageAge(prev, person) {
+    const age = person.died - person.born;
+
+    return prev + age;
+  }
+
```

Comment:
you have similar logic in several functions for calculating average age, move this function outside and reuse